### PR TITLE
Closes #1837: Update URL of Global Engagement link in global footer.

### LINKED ISSUE #1837: Update global.arizona.edu to international.arizona.edu  everywhere.
## Conditions of satisfaction
All instances of global.arizona.edu should be changed to international.arizona.edu 

Possibly a database update should be added to perform a string replacement.

### PR DESCRIPTION
**Pull request by joeparsons**
## Description
Updates URL of "Global Engagement" link in `az_global_footer` topics menu.  (Changed from https://global.arizona.edu to https://international.arizona.edu).

Adds DB update to update link on existing sites.

## Related issues
Closes #1837

## How to test
- Verify that the Global Engagement link in the global footer points at https://international.arizona.edu after installing the `az_global_footer` module on a new site (e.g. on ProboCI)
- Apply PR as patch to existing Quickstart site already using the `az_global_footer` module and ensure Global Engagement link URL is updated after running DB updates.

### Arizona Quickstart (install profile, custom modules, custom theme)
- Patch release changes
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [x] Critical institutional link or brand change
- Minor release changes
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [ ] Other or unknown

### Drupal core
- Patch release changes
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major or minor level update
- [ ] Other or unknown

### Drupal contrib projects
- Patch release changes
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major level update
- [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by joeparsons** [comment]
Right now the DB update will only update menu links in the `az-footer-topics` menu.  We could consider removing that condition if we want to update _any_ Drupal menu link pointed at `https://global.arizona.edu` with `https://international.arizona.edu`.

**Review by trackleft — APPROVED** [review]

File: modules/custom/az_global_footer/data/az_global_footer.json [hunk 1]
```diff
@@ -247,7 +247,7 @@
       "parent_link_id": "0",
       "menu": "az-footer-topics",
       "title": "Global Engagement",
```
Comment (camikazegreen) [inline modules/custom/az_global_footer/data/az_global_footer.json:249]: Update from Matt Littler in Arizona International.
```suggestion
      "title": "International Engagement",
```

**Review by camikazegreen — DISMISSED** [review]

File: modules/custom/az_global_footer/az_global_footer.install [hunk 2]
```diff
@@ -35,3 +36,26 @@ function az_global_footer_uninstall() {
     $executable->rollback();
   }
 }
+
+/**
+ * Update URL for Global Engagement link(s) in the global footer.
```
Comment (joeparsons) [inline modules/custom/az_global_footer/az_global_footer.install:41]:
```suggestion
 * Replace Global Engagement link(s) in the global footer with International Engagement link(s).
```

File: modules/custom/az_global_footer/az_global_footer.install [hunk 3]
```diff
@@ -35,3 +36,26 @@ function az_global_footer_uninstall() {
...
+ */
+function az_global_footer_update_920501() {
+  $updated_count = 0;
+  $menu_link_content_ids = \Drupal::entityQuery('menu_link_content')
+    ->condition('menu_name', 'az-footer-topics')
```
Comment (joeparsons) [inline modules/custom/az_global_footer/az_global_footer.install:46]:
```suggestion
    ->condition('menu_name', 'az-footer-topics')
    ->condition('title', 'Global Engagement')
```

File: modules/custom/az_global_footer/az_global_footer.install [hunk 4]
```diff
@@ -35,3 +36,26 @@ function az_global_footer_uninstall() {
...
+    ->condition('link__uri', 'https://global.arizona.edu')
+    ->execute();
+
+  foreach ($menu_link_content_ids as $id) {
+    /** @var \Drupal\menu_link_content\Entity\MenuLinkContent $menu_link_content */
+    $menu_link_content = MenuLinkContent::load($id);
+    if ($menu_link_content !== NULL) {
+      $menu_link_content->link__uri = 'https://international.arizona.edu';
```
Comment (joeparsons) [inline modules/custom/az_global_footer/az_global_footer.install:54]:
```suggestion
      $menu_link_content->title = 'International Engagement';
      $menu_link_content->link__uri = 'https://international.arizona.edu';      
```

File: modules/custom/az_global_footer/az_global_footer.install [hunk 5]
```diff
@@ -35,3 +36,28 @@ function az_global_footer_uninstall() {
...
+    $menu_link_content = MenuLinkContent::load($id);
+    if ($menu_link_content !== NULL) {
+      $menu_link_content->title = 'International Engagement';
+      $menu_link_content->link__uri = 'https://international.arizona.edu';      
+      $menu_link_content->save();
+      $updated_count++;
+    }
+  }
+
+  return t('Updated %count global footer menu link(s) with new Global Engagement URL (https://international.arizona.edu).', ['%count' => $updated_count]);
```
Comment (joeparsons) [inline modules/custom/az_global_footer/az_global_footer.install:62]:
```suggestion
  return t('Replaced %count Global Engagement menu link(s) in the global footer with International Engagement menu link(s).', ['%count' => $updated_count]);
```

**Comment by joeparsons** [comment]
`phpstan-drupal` couldn't figure out the properties on the link field on the MenuLinkContent entity:
```
lando phpstan
 ------ -------------------------------------------------------------------------------------------- 
  Line   web/profiles/custom/az_quickstart/modules/custom/az_global_footer/az_global_footer.install  
 ------ -------------------------------------------------------------------------------------------- 
  56     Access to an undefined property Drupal\link\LinkItemInterface::$uri.                        
 ------ -------------------------------------------------------------------------------------------- 
                                                                                                                        
 [ERROR] Found 1 error                                                                                                  
                                                                                                                        
```

So I added a single instance of `/* @phpstan-ignore-next-line */`

**Comment by joeparsons** [comment]
Tested on a dev version of the quickstart.arizona.edu site:
https://azd-1837-2-azs-quickstart.pantheonsite.io/

- Installed Global Footer module
- Applied PR diff as composer patch for `az-digital/az_quickstart`
- Ran database updates
- Verified that "Global Engagement" link in footer was updated to "International Engagement"

```
# terminus drush azs-quickstart.azd-1837-2 -- updatedb                                            
 [warning] This environment is in read-only Git mode. If you want to make changes to the codebase of this site (e.g. updating modules or plugins), you will need to toggle into read/write SFTP mode first.
 ------------------ ----------- --------------- --------------------------- 
  Module             Update ID   Type            Description                
 ------------------ ----------- --------------- --------------------------- 
  az_global_footer   920501      hook_update_n   920501 - Replace Global    
                                                 Engagement link(s) in the  
                                                 global footer with         
                                                 International Engagement   
                                                 link(s).                   
 ------------------ ----------- --------------- --------------------------- 


 Do you wish to run the specified pending updates? (yes/no) [yes]:
 > y

>  [notice] Update started: az_global_footer_update_920501
>  [notice] Replaced 1 Global Engagement menu link(s) in the global footer with International Engagement menu link(s).
>  [notice] Update completed: az_global_footer_update_920501
 [success] Finished performing updates.
```

This should be ready to merge now.

**Review by tadean — APPROVED** [review]

**Review by bberndt-uaz — APPROVED** [review]

**Review by trackleft — APPROVED** [review]

**Pull request merged**